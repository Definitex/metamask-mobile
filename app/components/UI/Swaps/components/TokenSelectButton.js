import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { View, StyleSheet, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import { colors } from '../../../../styles/common';
import RemoteImage from '../../../Base/RemoteImage';
import Text from '../../../Base/Text';

const styles = StyleSheet.create({
	container: {
		backgroundColor: colors.grey000,
		paddingVertical: 8,
		paddingHorizontal: 10,
		borderRadius: 100,
		flexDirection: 'row',
		alignItems: 'center',
		justifyContent: 'center'
	},
	icon: {
		marginRight: 8,
		width: 24,
		height: 24
	},
	emptyIcon: {
		borderRadius: 12,
		backgroundColor: colors.grey200,
		alignItems: 'center',
		justifyContent: 'center'
	},
	tokenSymbol: {
		fontSize: 16,
		textAlign: 'center',
		textAlignVertical: 'center'
	},
	caretDown: {
		textAlign: 'right',
		color: colors.grey500,
		marginLeft: 10,
		marginRight: 5
	}
});

// eslint-disable-next-line import/no-commonjs
const ethLogo = require('../../../../images/eth-logo.png');

const EmptyIcon = props => <View style={[styles.icon, styles.emptyIcon]} {...props} />;

function TokenSelectButton({ icon, symbol, onPress, disabled }) {
	const renderIcon = useMemo(() => {
		if (symbol === 'ETH') {
			return <RemoteImage fadeIn source={ethLogo} style={styles.icon} />;
		} else if (symbol && icon) {
			return <RemoteImage fadeIn source={{ uri: icon }} style={styles.icon} />;
		} else if (symbol) {
			return (
				<EmptyIcon>
					<Text style={styles.tokenSymbol}>{symbol[0].toUpperCase()}</Text>
				</EmptyIcon>
			);
		}

		return <EmptyIcon />;
	}, [symbol, icon]);
	return (
		<TouchableOpacity onPress={onPress} disabled={disabled}>
			<View style={styles.container}>
				{renderIcon}
				<Text primary>{symbol || 'Select a token'}</Text>
				<Icon name="caret-down" size={18} style={styles.caretDown} />
			</View>
		</TouchableOpacity>
	);
}

TokenSelectButton.propTypes = {
	icon: PropTypes.string,
	symbol: PropTypes.string,
	onPress: PropTypes.func,
	disabled: PropTypes.bool
};

export default TokenSelectButton;
