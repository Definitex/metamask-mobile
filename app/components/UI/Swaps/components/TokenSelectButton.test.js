import React from 'react';
import { shallow } from 'enzyme';
import TokenSelectButton from './TokenSelectButton';

describe('TokenSelectButton component', () => {
	it('should Render correctly', () => {
		const dummyHandler = jest.fn();
		const empty = shallow(<TokenSelectButton />);
		expect(empty).toMatchSnapshot();
		const eth = shallow(<TokenSelectButton symbol="ETH" />);
		expect(eth).toMatchSnapshot();
		const symbol = shallow(<TokenSelectButton symbol="cDAI" />);
		expect(symbol).toMatchSnapshot();
		const icon = shallow(
			<TokenSelectButton
				symbol="DAI"
				icon="https://cloudflare-ipfs.com/ipfs/QmNYVMm3iC7HEoxfvxsZbRoapdjDHj9EREFac4BPeVphSJ"
			/>
		);
		expect(icon).toMatchSnapshot();
		const onPress = shallow(
			<TokenSelectButton
				symbol="DAI"
				icon="https://cloudflare-ipfs.com/ipfs/QmNYVMm3iC7HEoxfvxsZbRoapdjDHj9EREFac4BPeVphSJ"
				onPress={dummyHandler}
			/>
		);
		expect(onPress).toMatchSnapshot();
	});
});
