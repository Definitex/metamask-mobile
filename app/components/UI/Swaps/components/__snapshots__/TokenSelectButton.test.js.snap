// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TokenSelectButton component should Render correctly 1`] = `
<TouchableOpacity>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f2f3f4",
        "borderRadius": 100,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 10,
        "paddingVertical": 8,
      }
    }
  >
    <EmptyIcon />
    <Text
      bold={false}
      centered={false}
      disclaimer={false}
      green={false}
      link={false}
      modal={false}
      primary={true}
      reset={false}
      right={false}
      small={false}
      strikethrough={false}
      upper={false}
    >
      Select a token
    </Text>
    <Icon
      allowFontScaling={false}
      name="caret-down"
      size={18}
      style={
        Object {
          "color": "#6a737d",
          "marginLeft": 10,
          "marginRight": 5,
          "textAlign": "right",
        }
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`TokenSelectButton component should Render correctly 2`] = `
<TouchableOpacity>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f2f3f4",
        "borderRadius": 100,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 10,
        "paddingVertical": 8,
      }
    }
  >
    <RemoteImage
      fadeIn={true}
      source={
        Object {
          "testUri": "../../../app/images/eth-logo.png",
        }
      }
      style={
        Object {
          "height": 24,
          "marginRight": 8,
          "width": 24,
        }
      }
    />
    <Text
      bold={false}
      centered={false}
      disclaimer={false}
      green={false}
      link={false}
      modal={false}
      primary={true}
      reset={false}
      right={false}
      small={false}
      strikethrough={false}
      upper={false}
    >
      ETH
    </Text>
    <Icon
      allowFontScaling={false}
      name="caret-down"
      size={18}
      style={
        Object {
          "color": "#6a737d",
          "marginLeft": 10,
          "marginRight": 5,
          "textAlign": "right",
        }
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`TokenSelectButton component should Render correctly 3`] = `
<TouchableOpacity>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f2f3f4",
        "borderRadius": 100,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 10,
        "paddingVertical": 8,
      }
    }
  >
    <EmptyIcon>
      <Text
        bold={false}
        centered={false}
        disclaimer={false}
        green={false}
        link={false}
        modal={false}
        primary={false}
        reset={false}
        right={false}
        small={false}
        strikethrough={false}
        style={
          Object {
            "fontSize": 16,
            "textAlign": "center",
            "textAlignVertical": "center",
          }
        }
        upper={false}
      >
        C
      </Text>
    </EmptyIcon>
    <Text
      bold={false}
      centered={false}
      disclaimer={false}
      green={false}
      link={false}
      modal={false}
      primary={true}
      reset={false}
      right={false}
      small={false}
      strikethrough={false}
      upper={false}
    >
      cDAI
    </Text>
    <Icon
      allowFontScaling={false}
      name="caret-down"
      size={18}
      style={
        Object {
          "color": "#6a737d",
          "marginLeft": 10,
          "marginRight": 5,
          "textAlign": "right",
        }
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`TokenSelectButton component should Render correctly 4`] = `
<TouchableOpacity>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f2f3f4",
        "borderRadius": 100,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 10,
        "paddingVertical": 8,
      }
    }
  >
    <RemoteImage
      fadeIn={true}
      source={
        Object {
          "uri": "https://cloudflare-ipfs.com/ipfs/QmNYVMm3iC7HEoxfvxsZbRoapdjDHj9EREFac4BPeVphSJ",
        }
      }
      style={
        Object {
          "height": 24,
          "marginRight": 8,
          "width": 24,
        }
      }
    />
    <Text
      bold={false}
      centered={false}
      disclaimer={false}
      green={false}
      link={false}
      modal={false}
      primary={true}
      reset={false}
      right={false}
      small={false}
      strikethrough={false}
      upper={false}
    >
      DAI
    </Text>
    <Icon
      allowFontScaling={false}
      name="caret-down"
      size={18}
      style={
        Object {
          "color": "#6a737d",
          "marginLeft": 10,
          "marginRight": 5,
          "textAlign": "right",
        }
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`TokenSelectButton component should Render correctly 5`] = `
<TouchableOpacity
  onPress={[MockFunction]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f2f3f4",
        "borderRadius": 100,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 10,
        "paddingVertical": 8,
      }
    }
  >
    <RemoteImage
      fadeIn={true}
      source={
        Object {
          "uri": "https://cloudflare-ipfs.com/ipfs/QmNYVMm3iC7HEoxfvxsZbRoapdjDHj9EREFac4BPeVphSJ",
        }
      }
      style={
        Object {
          "height": 24,
          "marginRight": 8,
          "width": 24,
        }
      }
    />
    <Text
      bold={false}
      centered={false}
      disclaimer={false}
      green={false}
      link={false}
      modal={false}
      primary={true}
      reset={false}
      right={false}
      small={false}
      strikethrough={false}
      upper={false}
    >
      DAI
    </Text>
    <Icon
      allowFontScaling={false}
      name="caret-down"
      size={18}
      style={
        Object {
          "color": "#6a737d",
          "marginLeft": 10,
          "marginRight": 5,
          "textAlign": "right",
        }
      }
    />
  </View>
</TouchableOpacity>
`;
